### ISSUE TITLE
Question: why to order by a field, it needs to be in the select?

### ISSUE THREAD

**Issue by marciogoularte** [mikependon/RepoDb#963]
Hi,

I'm using SQL Server and I have queries that make certain fields select and order by another

exemple:
`base.Query<TEntity>(TableName(), fields: Field.From("Id", "Name"), orderBy: new List<OrderField>() { new OrderField("Created", Order.Descending) });`

Throw exception here:
https://github.com/mikependon/RepoDB/blob/master/RepoDb.Core/RepoDb/StatementBuilders/BaseStatementBuilder.cs#L727

because without that it works too. What is the reason for this?

**Comment by mikependon** [comment]
We purposely have added this validation when composing a SQL statement.

Have you tried including the `Created` column from the list of fields you are querying? Technically, SQL Server could not order your result without it being listed as part of the selected columns.

**Comment by mikependon** [comment]
In short, the query below is not feasible.

```
SELECT Id, Name FROM TableName ORDER BY Created ASC;
```

But the query below would work.

```
SELECT Id, Name, Created FROM TableName ORDER BY Created ASC;
```

EDITED: Or, maybe, I am combining the sub query selections and other providers :smile:

**Comment by marciogoularte** [comment]
In my test I don't need Created (datetime). But I need to get Id and Name of the last or first users sorted by Created (datetime)

![image](https://user-images.githubusercontent.com/215987/138130272-06cd6a07-8734-440c-882b-2030fcc290d4.png)
ignore the order of IDs because the created of id 3 was changed to old date
![image](https://user-images.githubusercontent.com/215987/138130389-55bb141f-10c4-477d-ba89-c75873651b70.png)

SQL Server sorts even if the column is not part of the select

with Created
![image](https://user-images.githubusercontent.com/215987/138131603-55765afe-a2a6-4d3a-90dd-44a566143243.png)
![image](https://user-images.githubusercontent.com/215987/138131901-7bc0b36d-c8db-4345-9e2c-df65e63eb786.png)

but I don't need this column. I only need the Id and Name, but the order has to be the newest or the oldest

**Comment by mikependon** [comment]
Right, this is really a bug on the code. We will issue a fix for you, but we are glad if you can PR this simple fix.

It sounds like we need to remove the line [726](https://github.com/mikependon/RepoDB/blob/b79742ea46afb7bce8f241715bbe1a122cb1808a/RepoDb.Core/RepoDb/StatementBuilders/BaseStatementBuilder.cs#L726)-[740](https://github.com/mikependon/RepoDB/blob/b79742ea46afb7bce8f241715bbe1a122cb1808a/RepoDb.Core/RepoDb/StatementBuilders/BaseStatementBuilder.cs#L740), and also, the one in the `QueryAll` operation at line [792](https://github.com/mikependon/RepoDB/blob/b79742ea46afb7bce8f241715bbe1a122cb1808a/RepoDb.Core/RepoDb/StatementBuilders/BaseStatementBuilder.cs#L792)-[806](https://github.com/mikependon/RepoDB/blob/b79742ea46afb7bce8f241715bbe1a122cb1808a/RepoDb.Core/RepoDb/StatementBuilders/BaseStatementBuilder.cs#L806). It might affect some Unit Test, but the one we can always adjust to guard this logic.

Once fixed (either via PR or us), we will issue the release for you immediately.

Looking forward if you can PR this one mate.

**Comment by mikependon** [comment]
@SergerGood made the PR for this. The fix to this will be available the next releases > [RepoDb v1.12.10-beta1](https://www.nuget.org/packages/RepoDb/1.12.10-beta1).

**Comment by mikependon** [comment]
The fix for this is at PR #985
